Centralize conf

gturri opened the pull request:
Goals:
- There's only one conf file to change
- It seems we forgot the port in the html files. I added it. (correct me if I was wrong)

Next steps would be:
- Give the ability to have a `config.local.js` which could override the default `config.js` and be gitignored
- Let "http://" be in the config file (I've let it outside because it seems to mess things up because of slashes which are replaced by their ascii code. (probable just a mustache flag, I just didn't had the time to look into it yet)

gturri committed:
Gather the config in a single file

Nb: After renumeroting the numbered list in the README a few times, it seemed
wiser to choose the lazy way ("It’s important to note that the actual numbers
you use to mark the list have no effect on the HTML output Markdown produces")

AlexFaraino commented:
As this is a tiny application, many will use it behind a reverse proxy. Meaning, the port node listens on won't be the same the client will connect to.
i.e. nodejs listens on 8081 and a reverse proxy (Apache + mod_proxy, HA_proxy, Varnish, ...) listens on 80 and forwards requests for "door.mydomain.tld" to localhost:8081.
In this case, we can add two params in config file:
- `reverse-proxy-enable` (= 0 or `false` by default)
- `reverse-proxy-port` (=80 by default)

Great Job BTW. Thx

AlexFaraino commented:
And about the config files, we can just set the defaults in config.js.template. If config.js doesn't exists, nodejs can copy the template to config.js and tell users to make changes in this file.

AlexFaraino commented on index.js:
```diff
@@ -1,19 +1,32 @@
 var app = express();
-var port = 8081;
+var port = config.port;
+var url = config.host + ":" + port;
```
Here, the url can be:
`var url = config.host + ":" + (config.reverse-proxy ? config.reverse-proxy-port : port);`

AlexFaraino commented on config.js:
```diff
@@ -0,0 +1,7 @@
+
+config.port = 8081;
+config.host = "door.fabelier.org"
+
```
To support reverse proxy:
- `config.reverse-proxy = 0;`
- `config.reverse-proxy-port = 80;`

gturri commented:
Thanks for the explanation, I was unaware of the reverse proxy subtlety. Here is an updated of the PR with your suggestions.

gturri committed:
No more need to override versionned files with custom config

Seems easy to do asynchronous file copy with node, but I haven't found an easy way
to do it synchronously
(source of this code: http://procbits.com/2011/11/15/synchronous-file-copy-in-node-js)

gturri committed:
http:// is part of the config file

By default mustache escapes slashes, hence the need to override its escape method

gturri committed:
Made the logo url configurable
